Adding docutext and fixing usage of target

kedark3 opened the pull request:
**- What this PR does and why is it needed**
While reading through the readiness probes, I found it odd to call "version" as to check the readiness of nbctl and northd. But checking with @trozet @dcbw helped understand it is still valid and acceptable to check the version to determine readiness of the pods, because `ovn-appctl` still connecting to the app/daemon to determine the version. To remove the confusion, specially for readability to new contributors, it could be useful to have some comments there. 

In addition to that, I found that the `target` parameter was being not used/passed to readiness probe function correctly. Fixing that too. 

**- Special notes for reviewers**
This is my 1st PR to this repo. Hoping to get your kind reviews and get it merged. 

**- How to verify it**
I was able to run local development env using `./contrib/kind.sh` pre and post making these changes and pods came up correctly. Also verified if the changes I made to daemonset template took effect on launching cluster after I made changes. 
Happy to run any additional tests.


Signed-off-by:  Kedar Kulkarni <kkulkarni@redhat.com>

kedark3 commented:
Quay seems down, https://status.quay.io/ atm. Failing CI: 
![image](https://user-images.githubusercontent.com/8180469/112056808-d7c99180-8b2e-11eb-8dac-3feb26746c62.png)

kedark3 commented:
/retest

dcbw commented on go-controller/cmd/ovn-kube-util/app/readiness-probe.go:
```diff
@@ -103,15 +103,19 @@ func ovnSBDBReadiness(target string) error {
 }
 
 func ovnNorthdReadiness(target string) error {
+	// checking version works as it conntects to northd and returns the version
```
"connects"

dcbw commented on go-controller/cmd/ovn-kube-util/app/readiness-probe.go:
```diff
@@ -103,15 +103,19 @@ func ovnSBDBReadiness(target string) error {
-func ovnNbCtldReadiness(target string) error {
-	_, _, err := util.RunOVNAppctlWithTimeout(5, "-t", "ovn-nbctl", "version")
+func ovnNbCtlReadiness(target string) error {
+	// checking version works as it conntects to nbctld and returns the version
```
"connects to the ovn-nbctl daemon and returns"

kedark3 replied:
Right, it definitely confused me initially as well. I had to read up about it. I have added a note. PTAL. :)

kedark3 commented:
@dcbw I addressed your comments :+1: The CI failure seems to be flaky, nothing has changed from yesterday, apart from the comments. It had all passed otherwise. Happy to retest if you think necessary.

dcbw commented:
/lgtm
